sensor/kubernetes/enforcer/cronjob/suspend.go

```diff
@@ -0,0 +1,73 @@
+package cronjob
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	kubernetesPkg "github.com/stackrox/rox/pkg/kubernetes"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/sensor/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	batchV1      = "batch/v1"
+	batchV1beta1 = "batch/v1beta1"
+	fieldManager = "StackRox"
+)
+
+func makePatch(deploymentInfo *central.DeploymentEnforcement, apiVersion string) ([]byte, metav1.PatchOptions, error) {
+	forcePatch := true
+
+	patch := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"name": deploymentInfo.GetDeploymentName(),
+		},
+		"kind":       deploymentInfo.GetDeploymentType(),
+		"apiVersion": apiVersion,
+		"spec": map[string]interface{}{
+			"suspend": true,
+		},
+	}
+	patchBytes, err := json.Marshal(patch)
+	if err != nil {
+		return nil, metav1.PatchOptions{}, err
+	}
+
+	options := metav1.PatchOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       deploymentInfo.GetDeploymentType(),
+			APIVersion: apiVersion,
+		},
+		FieldManager: fieldManager,
+		Force:        &forcePatch,
+	}
+
+	return patchBytes, options, nil
+}
+
+// Suspend suspends the cron job
+func Suspend(ctx context.Context, client kubernetes.Interface, deploymentInfo *central.DeploymentEnforcement) (err error) {
+	if ok, apiErr := utils.HasAPI(client, batchV1, kubernetesPkg.CronJob); ok && apiErr == nil {
```

Comment:
One thing was different in my proposal - handling `apiErr` with retries. Looks like we have different opinions https://github.com/stackrox/stackrox/pull/4892#discussion_r1123493773 , but when I stepped into the code `ServerResourcesForGroupVersion` I saw that it makes api request to k8s api server. Therefore, I suggest explicitly handling `apiErr` rather than assuming v1beta1:

```suggestion
	if ok, apiErr := utils.HasAPI(client, batchV1, kubernetesPkg.CronJob); apiErr != nil {
		return retry.MakeRetryable(apiErr)
	} else if ok {
```